### PR TITLE
added env functionality and tests

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -223,3 +223,35 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 	}
 	return h, nil
 }
+
+// GetEnv gets the necessary env variables to use
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: GetHostEnv

File: pkg/minikube/cluster/cluster.go [hunk 2]
```diff
@@ -223,3 +223,35 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	TCP_PREFIX := "tcp://"
+	PORT_DELIMETER := ":"
+	PORT := "2376"
+
+	envMap := map[string]string{
+    "DOCKER_TLS_VERIFY": "1",
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: Indentation

File: pkg/minikube/cluster/cluster.go [hunk 3]
```diff
@@ -223,3 +223,35 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	
+	ip, err := host.Driver.GetIP()
+	if err != nil {
+		return nil, err
+	}
+
+	TCP_PREFIX := "tcp://"
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: I don't think UPPERCASE_VARIABLES is the golang style of writing code...

File: pkg/minikube/tests/driver_mock.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ func (driver *MockDriver) Create() error {
 }
 
 func (driver *MockDriver) GetIP() (string, error) {
-	return driver.BaseDriver.GetIP()
+	return "192.168.99.107", nil
```
Comment (luxas) [inline pkg/minikube/tests/driver_mock.go]: Should this be a test constant?

File: cmd/minikube/cmd/env.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+const (
+	SHELL_PREFIX = "export "
```
Comment (dlorenc) [inline cmd/minikube/cmd/env.go]: Switch these to UpperCamelCase

File: cmd/minikube/cmd/env.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+	SHELL_SUFFIX = "\"\n"
+	SHELL_DELIMITER = "=\""
+)
+
+
+// envCmd represents the env command
+var envCmd = &cobra.Command{
+	Use:   "env",
```
Comment (dlorenc) [inline cmd/minikube/cmd/env.go]: Let's rename to "docker-env" to be clear.

File: pkg/minikube/cluster/cluster_test.go [hunk 7]
```diff
@@ -333,3 +333,22 @@ func TestGetCredsError(t *testing.T) {
...
+
+func TestGetHostEnv(t *testing.T) {
+	api := tests.NewMockAPI()
+	createHost(api, MachineConfig{})
+
+	envMap, err := GetHostEnv(api)
+	if err != nil {
+		t.Fatalf("Unexpected error getting env: %s", err)
+	}
+	if _, hasTlsKey := envMap["DOCKER_TLS_VERIFY"]; !hasTlsKey {
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster_test.go]: You could make this into a loop to shorten it a bit.

File: pkg/minikube/tests/driver_mock.go (hunk 4 above)
Comment (dlorenc) [inline pkg/minikube/tests/driver_mock.go]: Sorry Aaron, I was wrong when we chatted about this. We already have a way to set the IP at a per-test level. See here:
https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cluster/cluster_test.go#L314

File: cmd/minikube/cmd/env.go [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+			os.Exit(1)
+		}
+		fmt.Fprintln(os.Stdout, buildDockerEnvShellOutput(envMap))
+	},
+}
+
+func buildDockerEnvShellOutput(envMap map[string]string) string {
+	output := ""
+	for env_name, env_val := range envMap {
+		output += shellPrefix + env_name + shellDelimiter + env_val + shellSuffix
```
Comment (dlorenc) [inline cmd/minikube/cmd/env.go]: Nit: this line might be easier to read as:

output += fmt.Sprintf("export %s=%s\n", env_name, env_val)

then you could drop the constants.

**Comment by dlorenc** [comment]
LGTM. Merging.

**Pull request merged**